client/signup/steps/site-information/index.jsx

```diff
@@ -51,125 +82,98 @@ class SiteInformation extends Component {
+					fieldDescription: translate(
+						"We'll use this as your site title. Don't worry, you can change this later."
+					),
+				};
+		}
+	}
 
-							{ isBusinessSiteSelected && (
-								<>
-									<FormFieldset>
-										<FormLabel htmlFor="address">
-											{ translate( 'Address' ) }
-											<InfoPopover className="site-information__info-popover" position="top">
-												{ translate( 'Where can people find your business?' ) }
-											</InfoPopover>
-										</FormLabel>
-										<FormTextarea
-											id="address"
-											name="address"
-											placeholder={ 'E.g. 60 29th Street #343\nSan Francisco, CA\n94110' }
-											onChange={ this.handleInputChange }
-											value={ this.state.address }
-										/>
-									</FormFieldset>
-									<FormFieldset>
-										<FormLabel htmlFor="email">
-											{ translate( 'Contact Email' ) }
-											<InfoPopover className="site-information__info-popover" position="top">
-												{ translate( 'Does your business have an email address?' ) }
-											</InfoPopover>
-										</FormLabel>
-										<FormTextInput
-											id="email"
-											name="email"
-											type="email"
-											placeholder={ 'E.g. email@domain.com' }
-											onChange={ this.handleInputChange }
-											value={ this.state.email }
-										/>
-									</FormFieldset>
-									<FormFieldset>
-										<FormLabel htmlFor="phone">
-											{ translate( 'Phone number' ) }
-											<InfoPopover className="site-information__info-popover" position="top">
-												{ translate( 'How can people contact you?' ) }
-											</InfoPopover>
-										</FormLabel>
-										<FormTelInput
-											id="phone"
-											name="phone"
-											placeholder={ translate( 'E.g. (555) 555-5555' ) }
-											onChange={ this.handleInputChange }
-											value={ this.state.phone }
-											pattern="*"
-										/>
-									</FormFieldset>
-								</>
-							) }
+	renderSubmitButton = () => (
+		<Button primary type="submit" onClick={ this.handleSubmit }>
+			{ this.props.translate( 'Continue' ) }
+		</Button>
+	);
 
-							<div className="site-information__submit-wrapper">
-								<Button primary type="submit" onClick={ this.handleSubmit }>
-									{ translate( 'Continue' ) }
-								</Button>
+	renderContent() {
+		const { hasMultipleFieldSets, informationFields } = this.props;
+		return (
+			<div
+				className={ classNames( 'site-information__wrapper', {
+					'has-multiple-fieldsets': hasMultipleFieldSets,
+				} ) }
+			>
+				<form>
+					{ informationFields.map( fieldName => {
+						const fieldTexts = this.getFieldTexts( fieldName );
+						const fieldIdentifier = `site-information__${ fieldName }`;
+						return (
+							<div
+								key={ fieldIdentifier }
+								className={ classNames( 'site-information__field-control', fieldIdentifier ) }
+							>
+								<FormFieldset>
```

Comment:
Hmmm, I created this PR with consideration to #29863, which creates a [Business-only flow](https://github.com/Automattic/wp-calypso/blob/ace9f752aa53f1a40c4abb82f18d7ff9c7158daa/client/signup/config/flows-pure.js#L117), and therefore doesn't demand any tricky internal config in the step component.

I'll find out whether that will be delayed. If so I can add a 'isBusiness' state to this component.

👍

Reply:
> Hmmm, I created this PR with consideration to #29863, which creates a Business-only flow, and therefore doesn't demand any tricky internal config in the step component.

Cool, that makes sense. 

> I'll find out whether that will be delayed. If so I can add a 'isBusiness' state to this component.

👍

Reply:
I've added the business-only check to the internal component state for now since the site-preview + style step has been slightly delayed. :)

Reply:
> Hmmm, I created this PR with consideration to #29863, which creates a Business-only flow, and therefore doesn't demand any tricky internal config in the step component.

Thinking through this again, I got slightly confused so request clarification :). Wouldn't we always show this step even for non-business flow since we need to show the site title field? So, would it then require a check to show or hide the address and phone field?

Reply:
> request clarification

You are right to. It's not straightforward, and I get tongue-twisted explaining it. 👍 

> Wouldn't we always show this step even for non-business flow since we need to show the site title field? 

The original motivation behind this PR was to allow us to split the site information fields into independent steps, and still support a unified site information form (one with all the steps on one page).

Originally, I was labouring under the over-optimistic assumption that we would be branching flows for business site types, [something that seems likely given this PR](https://github.com/Automattic/wp-calypso/pull/29863/files#diff-28b03eff6ef567c591024d6033cda11dR117). 

So, we would have one flow for Business whose steps would look something like this:
```
			steps: [
				'user',
				'site-type',
				'site-topic-with-preview',
				'site-style-with-preview',
				'site-information-title-with-preview',
				'site-information-address-with-preview',
				'site-information-phone-with-preview',
				'domains-with-preview',
				'plans',
			],
```
Because features such as 'site preview', 'site-style' and the extra site information fields were business-only, we could have gotten away with this. Other site types would have continued to use the `onboarding` flow, and a simple, `'site-information-title'` step.

Now it's not so certain. 

So you are right when you say we need a check to show or hide the address and phone fields, because, at least for now, the preview flow is on hold pending further design work. I've implemented this.

Please let me know if I can explain anything further (and more clearly)!

Reply:
Thanks for the explanation, that clarified things for me.